autonomy/cli/helpers/deployment.py

```diff
@@ -163,6 +168,127 @@ def run_deployment(
         stop_deployment(build_dir=build_dir)
 
 
+def _prepare_agent_env(working_dir: Path) -> None:
+    """Prepare agent env, add keys, run aea commands."""
+    env = json.loads((working_dir / "agent.json").read_text(encoding="utf-8"))
+    # Patch for trader agent
+    if "SKILL_TRADER_ABCI_MODELS_PARAMS_ARGS_STORE_PATH" in env:
+        data_dir = working_dir / "data"
+        data_dir.mkdir(exist_ok=True)
+        env["SKILL_TRADER_ABCI_MODELS_PARAMS_ARGS_STORE_PATH"] = str(data_dir)
+
+    # TODO: Dynamic port allocation, backport to service builder
+    env["CONNECTION_ABCI_CONFIG_HOST"] = "localhost"
+    env["CONNECTION_ABCI_CONFIG_PORT"] = "26658"
+
+    for var in env:
+        # Fix tendermint connection params
+        if var.endswith("MODELS_PARAMS_ARGS_TENDERMINT_COM_URL"):
+            env[var] = "http://localhost:8080"
+
+        if var.endswith("MODELS_PARAMS_ARGS_TENDERMINT_URL"):
+            env[var] = "http://localhost:26657"
+
+        if var.endswith("MODELS_PARAMS_ARGS_TENDERMINT_P2P_URL"):
+            env[var] = "localhost:26656"
+
+        if var.endswith("MODELS_BENCHMARK_TOOL_ARGS_LOG_DIR"):
+            benchmarks_dir = working_dir / "benchmarks"
+            benchmarks_dir.mkdir(exist_ok=True, parents=True)
+            env[var] = str(benchmarks_dir.resolve())
+
+    (working_dir / "agent.json").write_text(
+        json.dumps(env, indent=2),
+        encoding="utf-8",
+    )
+
+
+def _run_aea_cmd(
+    args: List[str],
+    cwd: Optional[Path] = None,
+    stdout: int = subprocess.PIPE,
+    stderr: int = subprocess.PIPE,
+    **kwargs: Any,
+) -> None:
+    """Run an aea command in a subprocess."""
+    result = subprocess.run(  # pylint: disable=subprocess-run-check # nosec
+        args=[sys.executable, "-m", "aea.cli", *args],
+        cwd=cwd,
+        stdout=stdout,
+        stderr=stderr,
+        **kwargs,
+    )
+    if result.returncode != 0:
+        std_error = result.stderr.decode()
+        if "Item with name ethereum already present!" not in std_error:
+            raise RuntimeError(f"Error running: {args} @ {cwd}\n{std_error}")
```

Comment:
What happens if this is the error?

Reply:
so the `Item with name ethereum already present!` error comes when we do `aea add-key ethereum` multiple times, this if was to skip that. I'm removing it and moving this setup call to build command because it makes more sense there